java/arcs/core/storage/ActiveStore.kt

```diff
@@ -49,17 +49,20 @@ abstract class ActiveStore<Data : CrdtData, Op : CrdtOperation, ConsumerData>(
     /** Handles a message from the storage proxy. */
     abstract suspend fun onProxyMessage(message: ProxyMessage<Data, Op, ConsumerData>): Boolean
 
-    override fun getStorageEndpoint(): StorageCommunicationEndpoint<Data, Op, ConsumerData> {
-        return object : StorageCommunicationEndpoint<Data, Op, ConsumerData> {
-            var id: Int? = null
+    /**
+     * Return a storage endpoint that will receive messages from the store via the
+     * provided callback
+     */
+    override fun getStorageEndpoint(
+        callback: ProxyCallback<Data, Op, ConsumerData>
+    ) = object : StorageCommunicationEndpoint<Data, Op, ConsumerData> {
+        val id = on(callback)
 
-            override fun setCallback(callback: ProxyCallback<Data, Op, ConsumerData>): Int =
-                on(callback).also { id = it }
+        override suspend fun onProxyMessage(
+            message: ProxyMessage<Data, Op, ConsumerData>
+        ) = this@ActiveStore.onProxyMessage(message.withId(id!!))
 
-            override suspend fun onProxyMessage(
-                message: ProxyMessage<Data, Op, ConsumerData>
-            ) = this@ActiveStore.onProxyMessage(message.withId(id!!))
-        }
+        override fun close() = off(id)
```

Comment:
Q: is this sufficient to release resources on the ServiceStore side/Storage layer side? It unregisters callbacks, but it doesn't unbindService() the service connection.

Reply:
Good question. You're right, I think that with this change, it also makes sense to to an unbind in the call to `off`. 

On this note: I'm considering additional follow-up changes related to this one:

* Changing `ActiveStore`'s `on`, `off`, and `onProxyMessage` to be protected, so that it's necessary to get a `StorageCommunicationEndpoint` to use it.
* Perhaps going as far as changing the `Store.activate` method to simply return a storage communication endpoint. 
* Modifying the StorageService AIDL interfaces to match the shape of the changes to these Kotlin interfaces (either calling bind with an IStorageServiceCallback, or creating a similar `StorageCommunicationEndpointProvider`/`StorageCommunicationEndpoint` interface pairs)

But these would all be changes that touch a lot more places, so I wanted to start the discussion with a much simpler piece of the puzzle.